include/rocksdb/utilities/transaction_db.h

```diff
@@ -458,6 +458,37 @@ class TransactionDB : public StackableDB {
   virtual std::vector<DeadlockPath> GetDeadlockInfoBuffer() = 0;
   virtual void SetDeadlockInfoBufferSize(uint32_t target_size) = 0;
 
+  // Caller must ensure there are no active writes when this API is called.
+  // Create a shared snapshot and assign ts to it.
+  virtual std::shared_ptr<const Snapshot> CreateSharedSnapshot(
```

Comment:
Need to explain what the function does. I guess it is to create a snapshot with this tx, which is tracked by the DB and is also returned?